### PR TITLE
First working state

### PR DESCRIPTION
**Pull request by jaronoff97**
Initializes the repo with a working binary that lets a customer run checks against their Kubernetes cluster

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      matrix:
+        # build and publish in parallel: linux/386, linux/amd64, linux/arm64, windows/386, windows/amd64, darwin/amd64, darwin/arm64
+        goos: [linux, windows, darwin]
+        goarch: ["386", amd64, arm64]
+        exclude:
+          - goarch: "386"
+            goos: darwin
+          - goarch: arm64
+            goos: windows
+    steps:
```
Comment (smithclay) [inline .github/workflows/release.yaml:19]: nice to have in the future as well: a docker image published to the project

Comment (jaronoff97) [reply .github/workflows/release.yaml:19]: so i was thinking this was going to be a binary that folks can install, but maybe eventually we could run this as an init container.

File: cmd/check.go [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+	dependencies []dependencies.Initializer
+	checkers     []checks.NewChecker
+}
+
+var (
+	kubeConfig      string
+	accessToken     string
+	http            bool
+	availableChecks = map[string]checkGroup{
+		"metrics": {
```
Comment (smithclay) [inline cmd/check.go:48]: optional (but I think every helpful) -- `grpcurl` (or similar) command to deal with proxy/firewall issues (does _any_ data make it out vs otel-specific data)

Comment (jaronoff97) [reply cmd/check.go:48]: ah yeah thats like a step between that's useful to have :) I could have a whole DNS check actually

**Review by smithclay — APPROVED** [review]
Mostly comments around possible other checks/future features. Nothing blocking.

**Review by jmacd — COMMENTED** [review]
LGTM[Approved]

**Pull request merged**